courses/views/v1/__init__.py

```diff
@@ -381,19 +382,6 @@ def respond(data, status=True):  # noqa: FBT002
         return resp
 
 
-@extend_schema(
```

Comment:
This was requiring a path parameter in places it shouldn't, like `list` and `create` urls. 

A different way to work around this is to use 
```py
@extend_schema_view(
    partial_update=extend_schema(...)
    update=extend_schema(...),
    destroy=extend_schema(...),
)
```

The original issue about why the path parameter wasn't being detected properly is described in `extend_schema_get_queryset` util function below.